### ISSUE TITLE
[Suggestion] Add /findbiome

### ISSUE THREAD

**Issue by ZenithRogue** [Earthcomputer/clientcommands#56]
Basically would just allow you to find the nearest instance of a particular biome. Pretty simple in concept (though probably not as simple in execution).

Since there are so many variations of the basic biomes, such as the Jungle with all `Jungle, Jungle Hills, Junge Edge, Modified Jungle, Modified Jungle Edge, Bamboo Jungle, Bamboo Jungle Hills`, it would be nice to have biome groups like `#Jungle, #Plains, #Mountains, etc`

**Comment by MCRcortex** [comment]
this is not even remotely simple as it would require cracking the world seed which is hard enough already

**Comment by hube12** [comment]
this mod to find a biome already exist and i dont think it should belong inside clientcommand

**Comment by User670** [comment]
/cfind, /cfinditem, /cfindblock all works by searching through client-side knowledge (all blocks and entities are loaded in nearby chunks, and chest items are accessed by automatically opening nearby chests). However, finding biomes requires searching through unloaded chunks, which is not information readily available on the client side.

**Comment by Earthcomputer** [comment]
The way I would imagine it working is to only search for biomes nearby the player. Which I find a little bit useless because you can often just look around and immediately tell which biomes are nearby.
From what I understand this request to be, it's more than that, and I would be going into the rabbithole of seed cracking by that point. It's not something I'm categorically saying no to, but it would definitely be an expansion of scope in clientcommands comparable to when I first added enchantment cracking.

**Comment by leumasme** [comment]
if this was added it would be massive, but i can understand if you say that this is not going to happen.

**Comment by ZenithRogue** [comment]
Yeah, seed cracking would be probably be difficult as hell, but I don't see any way around it for this particular request

**Comment by hube12** [comment]
well the obvious way is to provide the seed that has been found otherwise with an external tool and then it does the biome finding in situ, but since generating a biome map is quite ressource intensive, this doesnt make much sense inside the mod

**Comment by User670** [comment]
Oh another thing. Suppose you want to crack the seeds, what information would you use to do so? If you use blocks, keep in mind that terrain can be altered by a whole lot of things, eg. corrupted chunk data, admins and world editing tools, and heck even players digging/building. (I wonder what, other than slime chunks, can be dependent on seed and unaffected from external factors)

**Comment by hube12** [comment]
@user670 can we cut the seed cracking conversation, it's not the place to do it here, there is already plenty research on that and it's really easy. As for now this issue is not yet fully considered and so forth there is no need to go any deeper (but if you want to personally then click on my github and go on my youtube channel i guess)

**Comment by ImUrX** [comment]
@hube12 if the seed is already cracked just use chunkbase, its quite good for that. 
i think it would be nice that world cracking would be added into the mod and then make this implemented (i know its quite hard on the computer, but it would be nice to at least try an optimized version of both things)

**Comment by hube12** [comment]
@ImUrX the strain of the crack isn't really an issue, we already have wayw that takes a few seconds at top. Also i dont agree that chunkbase is quite good at it, it's below average speed wise but since it doesn't require to install an extra mod (minihud for instance) or program (amidst) i get it. Anyway, this issue has been considered, maybe one day ;)

**Comment by Earthcomputer** [comment]
More to the point of a `/findbiome` command, I can see it being useful if clientcommands somehow gets the world seed from somewhere. How it might get the world seed is beside the point, and if you want a built-in seed cracker, that belongs in a separate issue.

**Comment by ZenithRogue** [comment]
Made a new issue #85 to mitigate seed cracking discussion

**Comment by xpple** [comment]
Out of scope for clientcommands to implement using the client's code. There are other mods that can do this.
